lightning/src/ln/channelmanager.rs

```diff
@@ -289,6 +289,27 @@ struct ReceiveError {
 	msg: &'static str,
 }
 
+/// This enum is used to specify which error data to send to peers when failing back an HTLC
+/// using [`ChannelManager::fail_htlc_backwards_with_reason`].
+///
+/// For more info on failure codes, see <https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages>.
+#[derive(Clone, Copy)]
+pub enum FailureCode {
+	/// We had a temporary error processing the payment. Useful if no other error codes fit 
+	/// and you want to indicate that the payer may want to retry.
+	TemporaryNodeFailure             = 0x2000 | 2,
+	/// We have a required feature which was not in this onion. For example, you may require
+	/// some additional metadata that was not provided with this payment.
+	RequiredNodeFeatureMissing       = 0x4000 | 0x2000 | 3,
+	/// You may wish to use this when a `payment_hash` is unknown, the `payment_secret` is
+	/// required and not present, the `payment_secret` doesn't match the `payment_hash`, the
+	/// received amount is incorrect, the CLTV expiry of the HTLC is too close to the current
+	/// block height for safe handling, or `payment_metadata` isn't present when it should be.
```

Comment:
Ah so some of these checks will be completed by LDK on the user's behalf:
- the `payment_secret` is required and not present
- `payment_secret` doesn't match the `payment_hash`
- received amount is incorrect
- `payment_metadata` isn't present when it should be 

Would suggest removing those